### ISSUE TITLE
cRPD: Automatically load license

### ISSUE THREAD

**Issue by jrokeach** [srl-labs/containerlab#648]
https://github.com/srl-labs/containerlab/blob/0fae5d7d68413022ce93123d9988b694f7dafed6/nodes/crpd/crpd.go#L155

If you copy this to `/config/license/safenet/junos_sfnt.lic` instead of `/config/license.conf`, it will automatically load in cRPD rather than requiring something like a `clab exec -t clos-6crpd.clab.yml --cmd "cli request system license add /config/license.conf`.

**Comment by hellt** [comment]
Nice! that is exactly the platform knowledge we needed to enhance platforms. someone will implement this, as the change is rather trivial

**Comment by chadell** [comment]
@jrokeach , when trying to use this in crpd version 19.4R1.10 (the one available from Juniper for evaluation purposes), I'm getting this error:

```
sudo containerlab deploy -t topo-crpd.yml --reconfigure
INFO[0000] Containerlab v0.25.1 started                 
INFO[0000] Parsing & checking topology file: topo-crpd.yml 
INFO[0000] Removing /home/fedora/clab-test directory... 
INFO[0000] Creating lab directory: /home/fedora/clab-test 
ERRO[0000] failed pre-deploy phase for node "r1": file copy [src /home/fedora/crpd-license.key -> dst /home/fedora/clab-test/r1/config/license/safenet/junos_sfnt.lic] failed open /home/fedora/clab-test/r1/config/license/safenet/junos_sfnt.lic: no such file or directory 
```

Clab topology 

```yml
name: test
topology:
  nodes:
    r1:
      kind: crpd
      image: crpd:19.4R1.10
      license: crpd-license.key
```

would be this related to an older version of crpd?

**Comment by hellt** [comment]
Hi Christian,
Do you have this license file by the provided path?

**Comment by chadell** [comment]
> Hi Christian, Do you have this license file by the provided path?

Yes, I have it (the source one)

```
$ ls -ltr /home/fedora/crpd-license.key 
-rw-r--r--. 1 fedora fedora 1806 Apr 17 05:13 /home/fedora/crpd-license.key
```

From the error message `failed open /home/fedora/clab-test/r1/config/license/safenet/junos_sfnt.lic`, I guess if it could be something related to the folder structure created that is missing the `license` folder, and then the copy command fails.

```
$ ls -ltr /home/fedora/clab-test/r1/config/
total 8
-rw-r--r--. 1 root root 3398 Apr 17 17:34 sshd_config
-rw-r--r--. 1 root root  227 Apr 17 17:34 juniper.conf
```

I could try to better understand the error, but just wondering if maybe someone have tried this evaluation image, before digging in.

Also, when trying to run the image without the `license` parameter, it also complains about `ERRO[0000] failed deploy phase for node "r2": Error response from daemon: No command specified `

**Comment by jrokeach** [comment]
@chadell this isn't related to cRPD as the copy into the container isn't working in the first place. If the container was spun up without license it'd still run, though functionality would be crippled.

**Comment by LimeHat** [comment]
@chadell did you add this image via `docker import` from a tarball?
If so, please remove it and use `docker load` instead

PS @hellt can we add a big note to the website about this? This questions pops up every other week..

**Comment by hellt** [comment]
@LimeHat can add it to crpd man page, but chances are low that it will work flawlessly. I guess what would be nice to add is an error message in case an image that is about to be run has an empty entry point and cmd

**Comment by hellt** [comment]
@chadell i will check the license copy routine for crpd

**Comment by chadell** [comment]
thanks @LimeHat , that worked like a charm. I went with the `docker import` as it's the option I am used to use in 99% of cases, and forgot to try with the `load` one. I was able then to access the cli and import the license manually.

@hellt , if you could give at quick try it would be great, but if you need help I could spend some time debugging it. Thank you very much for so great tool! Containerlab is helping me a lot to build lab scenarios.

**Comment by hellt** [comment]
added a fix for this lic copy issue in #863
